test/controllers/UnloadingPermissionControllerSpec.scala

```diff
@@ -16,232 +16,279 @@
+        <CouCO125>IT</CouCO125>
+        <TINCO159>IT444100201000</TINCO159>
+      </TRACONCO1>
+      <TRACONCE1>
+        <NamCE17>Mancini Carriers</NamCE17>
+        <StrAndNumCE122>90 Desio Way</StrAndNumCE122>
+        <PosCodCE123>MOD 5JJ</PosCodCE123>
+        <CitCE124>Modena</CitCE124>
+        <CouCE125>IT</CouCE125>
+        <TINCE159>IT444100201000</TINCE159>
+      </TRACONCE1>
+      <TRADESTRD>
+        <NamTRD7>The Luggage Carriers</NamTRD7>
+        <StrAndNumTRD22>225 Suedopolish Yard,</StrAndNumTRD22>
+        <PosCodTRD23>SS8 2BB</PosCodTRD23>
+        <CitTRD24>,</CitTRD24>
+        <CouTRD25>GB</CouTRD25>
+        <TINTRD59>GB163910077000</TINTRD59>
+      </TRADESTRD>
+      <CUSOFFDEPEPT>
+        <RefNumEPT1>IT021100</RefNumEPT1>
+      </CUSOFFDEPEPT>
+      <CUSOFFPREOFFRES>
+        <RefNumRES1>GB000060</RefNumRES1>
+      </CUSOFFPREOFFRES>
+      <SEAINFSLI>
+        <SeaNumSLI2>3</SeaNumSLI2>
+        <SEAIDSID>
+          <SeaIdeSID1>Seals01</SeaIdeSID1>
+        </SEAIDSID>
+        <SEAIDSID>
+          <SeaIdeSID1>Seals02</SeaIdeSID1>
+        </SEAIDSID>
+        <SEAIDSID>
+          <SeaIdeSID1>Seals03</SeaIdeSID1>
+        </SEAIDSID>
+      </SEAINFSLI>
+      <GOOITEGDS>
+        <IteNumGDS7>1</IteNumGDS7>
+        <GooDesGDS23>Flowers</GooDesGDS23>
+        <GroMasGDS46>1000</GroMasGDS46>
+        <NetMasGDS48>999</NetMasGDS48>
+        <CouOfDisGDS58>GB</CouOfDisGDS58>
+        <CouOfDesGDS59>GB</CouOfDesGDS59>
+        <PRODOCDC2>
+          <DocTypDC21>235</DocTypDC21>
+          <DocRefDC23>Ref.</DocRefDC23>
+        </PRODOCDC2>
+        <CONNR2>
+          <ConNumNR21>container 1</ConNumNR21>
+        </CONNR2>
+        <CONNR2>
+          <ConNumNR21>container 2</ConNumNR21>
+        </CONNR2>
+        <PACGS2>
+          <MarNumOfPacGS21>Ref.</MarNumOfPacGS21>
+          <KinOfPacGS23>BX</KinOfPacGS23>
+          <NumOfPacGS24>1</NumOfPacGS24>
+        </PACGS2>
+        <SGICODSD2>
+          <SenGooCodSD22>1</SenGooCodSD22>
+          <SenQuaSD23>1</SenQuaSD23>
+        </SGICODSD2>
+      </GOOITEGDS>
+      <GOOITEGDS>
+        <IteNumGDS7>1</IteNumGDS7>
+        <GooDesGDS23>Flowers</GooDesGDS23>
+        <GroMasGDS46>1000</GroMasGDS46>
+        <NetMasGDS48>999</NetMasGDS48>
+        <CouOfDisGDS58>GB</CouOfDisGDS58>
+        <CouOfDesGDS59>GB</CouOfDesGDS59>
+        <PRODOCDC2>
+          <DocTypDC21>235</DocTypDC21>
+          <DocRefDC23>Ref.</DocRefDC23>
+        </PRODOCDC2>
+        <CONNR2>
+          <ConNumNR21>container 1</ConNumNR21>
+        </CONNR2>
+        <CONNR2>
+          <ConNumNR21>container 2</ConNumNR21>
+        </CONNR2>
+        <PACGS2>
+          <MarNumOfPacGS21>Ref.</MarNumOfPacGS21>
+          <KinOfPacGS23>BX</KinOfPacGS23>
+          <NumOfPacGS24>1</NumOfPacGS24>
+        </PACGS2>
+        <SGICODSD2>
+          <SenGooCodSD22>1</SenGooCodSD22>
+          <SenQuaSD23>1</SenQuaSD23>
+        </SGICODSD2>
+      </GOOITEGDS>
+    </CC043A>
+
```

Comment:
Do we still need all of this test data setup if we're mocking the deserialisation dependencies?